riscv/mmu.cc

```diff
@@ -149,16 +149,31 @@ bool mmu_t::mmio_load(reg_t paddr, size_t len, uint8_t* bytes)
 {
   if (!mmio_ok(paddr, LOAD))
     return false;
-
-  return sim->mmio_load(paddr, len, bytes);
+  bool power_of_2 = (len & (len - 1)) == 0;
+  bool naturally_aligned = (paddr & (len - 1)) == 0;
+  if (power_of_2 && naturally_aligned)
+    return sim->mmio_load(paddr, len, bytes);
+  for (size_t i = 0; i < len; i++) {
+    if (!sim->mmio_load(paddr + i, 1, bytes + i))
+      return false;
+  }
+  return true;
```

Comment:
Can you think of any way to remove all the copypasta? These two methods are the same except for the `sim->mmio_*` method called.

Reply:
Sure, I’ll find a way to refactor it this weekend.

Reply:
I also realized that we want to call `mmio_ok` on each byte, since these accesses might span boundaries of MMIO-ok-ness.

Reply:
Done.  I'm not wild about the `const_cast` I ended up with, but it's reasonably contained.